app/server/src/main/scala/org/bitcoins/server/Main.scala

```diff
@@ -127,6 +145,133 @@ object Main extends App {
     }
   }
 
+  // todo move me
+  case class EclairBitcoindPair(
+      eclair: EclairRpcClient,
+      bitcoind: BitcoindRpcClient) {
+
+    require(eclair.instance.network == bitcoind.instance.network,
+            "Eclair and bitcoind must be on the same network")
+    require(eclair.instance.network == nodeConf.network,
+            "Eclair must be on the same network as us")
+    require(bitcoind.instance.network == nodeConf.network,
+            "bitcoind must be on the same network as us")
+    require(
+      eclair.instance.bitcoindAuthCredentials.get == bitcoind.instance.authCredentials,
+      "Eclair's bitcoind authCredentials must match bitcoind's")
+    require(
+      eclair.instance.bitcoindRpcUri.get == bitcoind.instance.rpcUri,
+      s"Eclair's bitcoind rpc uri must match bitcoind's, ${eclair.instance.bitcoindRpcUri.get} != ${bitcoind.instance.rpcUri}"
+    )
+    require(bitcoind.instance.uri.getHost == peerSocket.getHostString,
+            s"${bitcoind.instance.uri.getHost} != ${peerSocket.getHostString}")
+    require(bitcoind.instance.uri.getPort == peerSocket.getPort,
+            s"${bitcoind.instance.uri.getPort} != ${peerSocket.getPort}")
+    require(eclair.instance.zmqConfig.isDefined,
+            "Eclair must have a zmq config")
+
+    require(
+      eclair.instance.zmqConfig.get.rawBlock == bitcoind.instance.zmqConfig.rawBlock,
+      s"Eclair's bitcoind zmq raw block must match bitcoind's, ${eclair.instance.zmqConfig.get.rawBlock} != ${bitcoind.instance.zmqConfig.rawBlock}"
+    )
+    require(
+      eclair.instance.zmqConfig.get.rawTx == bitcoind.instance.zmqConfig.rawTx,
+      s"Eclair's bitcoind zmq raw tx must match bitcoind's, ${eclair.instance.zmqConfig.get.rawTx} != ${bitcoind.instance.zmqConfig.rawTx}"
+    )
+
+    def start: Future[EclairBitcoindPair] = {
+      val f: Runnable = new Runnable {
+        override def run(): Unit = {
+          eclair.start()
+          ()
+        }
+      }
+
+//      val startEclairF = eclair.start()
+      f.run()
+      val startBitcoindF = bitcoind.start()
+      for {
+        startedBitcoind <- startBitcoindF
+      } yield EclairBitcoindPair(eclair, startedBitcoind)
+    }
+
+    def stop: Future[EclairBitcoindPair] = {
+      val stoppedEclairF = eclair.stop()
```

Comment:
Do want to to stop these in parallel? We probably want to make sure eclair shuts down _completely_ before we stop bitcoind. This way we know that bitcoind can receive all information that eclair sends on shutdown.